content/en/docs/guides/flux-e2e.md

```diff
@@ -0,0 +1,148 @@
+
+`flux create source git --help`
+
+`flux create kustomization --help`
+
+These generators can be used interactively to imperatively create Flux resources in the cluster, or as preferred for GitOps: when called with the `--export` option, `flux create ...` can emit YAML on stdout, that can be captured in a file and committed to create the resource.
+
+For more information, see: [`flux create`](https://fluxcd.io/docs/cmd/flux_create/).
+
+Some resource options are not available through generators and can only be accessed through fields in YAML, (users are generally expected to write resources in YAML and commit them, and should do so when they require access to those features.) Flux's OpenAPI specification can also assist users in producing valid YAML for creating resources in Flux APIs.
+
+### 2. `flux diff kustomization`
+
+Users have an opportunity to inspect the change of a repository from the Flux CLI, ahead of where Flux actually applies it to the cluster. This new feature landed in Flux 0.26.
+
+Run `flux diff kustomization --path=./clusters/my-cluster flux-system` from the bootstrap repo, or point it at any other Flux Kustomization and the matching path in your configuration repository to observe what changes Flux will apply, even before they are committed and pushed. This takes account of the cluster state and so it can also be used at any time to check for drift on the cluster that Flux would revert back to the state in Git as soon as the Kustomization is reconciled, or at its next interval.
+
+Any diff containing secret data is obscured so that no secret data is accidentally divulged from the diff output.
+
+### 3. `ImageRepository` and `ImageUpdateAutomation` with `ImagePolicy`
+
+Flux can create git commits to apply updates to the cluster, that are applied in the standard GitOps way to the cluster (as a git commit), written by a Flux agent called Image Automation Controller. It works through these resources with the help of the Image Reflector Controller to determine when updates are available and apply them to the cluster.
+
+### 4. `git commit`
+
+Except when it happens as a result of Image Automation, the commit itself happens outside of Flux's purview, so the commit event itself has no effect on remote systems until it is pushed.
+
+When the cluster reconciles a Source resource (like `GitRepository` or `Bucket`) the content in the new revision is captured on the cluster through a set of filters (sourceignore, spec.ignore, ...) and collected in a tar file to be stored; this file is known as an Artifact in Flux, and it can be accessed by any agent in the `flux-system` namespace.
+
+When pushed, the receipt of a new commit activates the Git host to fire a webhook to notify subscribers about a `push` event, which Flux can consume via its `Receiver` API.
+
+### 4.5 `NetworkPolicy` and the `flux-system` namespace
```

Comment:
This section is more suitable in the install docs, where we can explain what `flux install/bootstrap --network-policy` does.